nilearn/image/image.py

```diff
@@ -1012,21 +1043,30 @@ def threshold_img(
             )
 
         mask_data, _ = masking.load_mask_img(mask_img)
+
+        # Take only points that are within the mask to check for threshold
+        img_data_for_cutoff = img_data_for_cutoff[mask_data != 0.0]
+
         # Set as 0 for the values which are outside of the mask
         img_data[mask_data == 0.0] = 0.0
 
     cutoff_threshold = check_threshold(
         threshold,
-        img_data,
+        img_data_for_cutoff,
         percentile_func=scoreatpercentile,
         name="threshold",
+        two_sided=two_sided,
     )
 
     # Apply threshold
     if two_sided:
-        img_data[np.abs(img_data) < cutoff_threshold] = 0.0
+        img_data[
+            (-cutoff_threshold <= img_data) & (img_data <= cutoff_threshold)
```

Comment:
```suggestion
            (img_data => -cutoff_threshold) & (img_data <= cutoff_threshold)
```

Reply:
I guess you were trying to get to something like

- `-cutoff_threshold <= img_data <= cutoff_threshold`

I kinda of prefer to keep the "fixed" part of a comparison.

But in this case I am not sure which one is more readable.

Reply:
I think the way I did it is more readable. But if you prefer otherwise, I can commit your suggestion. Let me know.

Reply:
na it's ok